tests/nightly/test_large_vector.py

```diff
@@ -708,6 +708,182 @@ def test_full():
+
+
+def test_round():
+    x = create_input_for_rounding_ops()
+    y = nd.round(x)
+    assert y[LARGE_X//2-2] == -1
+    assert y[LARGE_X//2-1] == -1
+    assert y[LARGE_X//2] == 0
+    assert y[LARGE_X//2+1] == 1
+    assert y[LARGE_X//2+2] == 1
+
+
+def test_trunc():
+    x = create_input_for_rounding_ops()
+    y = nd.trunc(x)
+    assert y[LARGE_X//2-2] == -1
+    assert y[LARGE_X//2-1] == 0
+    assert y[LARGE_X//2] == 0
+    assert y[LARGE_X//2+1] == 0
+    assert y[LARGE_X//2+2] == 1
+
+
+def test_arcsin():
+    x = nd.array([-1, -.707, 0, .707, 1])
+    x = nd.tile(x, LARGE_X//5)
+    y = nd.arcsin(x)
+    assert_almost_equal(y[0].asnumpy(), -np.pi/2, atol=1e-3)
+    assert_almost_equal(y[1].asnumpy(), -np.pi/4, atol=1e-3)
+    assert_almost_equal(y[-3].asnumpy(), 0, atol=1e-3)
+    assert_almost_equal(y[-2].asnumpy(), np.pi/4, atol=1e-3)
+    assert_almost_equal(y[-1].asnumpy(), np.pi/2, atol=1e-3)
+
+
+def test_arccos():
+    x = nd.array([-1, -.707, 0, .707, 1])
+    x = nd.tile(x, LARGE_X//5)
+    y = nd.arccos(x)
+    assert_almost_equal(y[0].asnumpy(), np.pi, atol=1e-3)
+    assert_almost_equal(y[1].asnumpy(), 3*np.pi/4, atol=1e-3)
+    assert_almost_equal(y[-3].asnumpy(), np.pi/2, atol=1e-3)
+    assert_almost_equal(y[-2].asnumpy(), np.pi/4, atol=1e-3)
+    assert_almost_equal(y[-1].asnumpy(), 0, atol=1e-3)
+
+
+def test_arctan():
+    x = nd.array([-np.Inf, -1, 0, 1, np.Inf])
+    x = nd.tile(x, LARGE_X//5)
+    y = nd.arctan(x)
+    assert_almost_equal(y[0].asnumpy(), -np.pi/2, atol=1e-3)
+    assert_almost_equal(y[1].asnumpy(), -np.pi/4, atol=1e-3)
+    assert_almost_equal(y[-3].asnumpy(), 0, atol=1e-3)
+    assert_almost_equal(y[-2].asnumpy(), np.pi/4, atol=1e-3)
+    assert_almost_equal(y[-1].asnumpy(), np.pi/2, atol=1e-3)
+
+
+def test_sin():
+    x = nd.array([-np.pi/2, -np.pi/4, 0, np.pi/4, np.pi/2])
+    x = nd.tile(x, LARGE_X//5)
+    y = nd.sin(x)
+    assert_almost_equal(y[0].asnumpy(), -1, atol=1e-3)
+    assert_almost_equal(y[1].asnumpy(), -.707, atol=1e-3)
+    assert_almost_equal(y[-3].asnumpy(), 0, atol=1e-3)
+    assert_almost_equal(y[-2].asnumpy(), .707, atol=1e-3)
+    assert_almost_equal(y[-1].asnumpy(), 1, atol=1e-3)
+
+
+def test_cos():
+    x = nd.array([0, np.pi/4, np.pi/2, 3*np.pi/4, np.pi])
+    x = nd.tile(x, LARGE_X//5)
+    y = nd.cos(x)
+    assert_almost_equal(y[0].asnumpy(), 1, atol=1e-3)
+    assert_almost_equal(y[1].asnumpy(), .707, atol=1e-3)
+    assert_almost_equal(y[-3].asnumpy(), 0, atol=1e-3)
+    assert_almost_equal(y[-2].asnumpy(), -.707, atol=1e-3)
+    assert_almost_equal(y[-1].asnumpy(), -1, atol=1e-3)
+
+
+def test_tan():
+    x = nd.array([-np.pi/4, 0, np.pi/4])
+    x = nd.tile(x, LARGE_X//3)
+    y = nd.tan(x)
+    assert y[0] == -1
+    assert y[1] == 0
+    assert y[-1] == 1
+
+
+def test_radians():
+    x = nd.array([0, 90, 180, 270, 360])
+    x = nd.tile(x, LARGE_X//5)
+    y = nd.radians(x)
+    assert_almost_equal(y[0].asnumpy(), 0, atol=1e-3)
```

Comment:
dont both parameters have to be numpy ndarrays ?

Reply:
works for both ndarrays and numpy arrays: https://github.com/apache/incubator-mxnet/blob/c2bbde75c36be2fef4a5e6d941cc01f5aabd9fd2/python/mxnet/test_utils.py#L540

Reply:
yes.... this implementation worked too I can change it to following instead 
`assert np.abs(y[0].asnumpy(), 0) <= 1e-3`
doyou think this is better ?

Reply:
its fine if it works.